test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/CredentialsTests.kt

```diff
@@ -88,29 +88,31 @@ class CredentialsTests {
         assertFailsWith<IllegalArgumentException> { Credentials.emailPassword("foo@bar.com", "") }
 
         // Test Google as a special case as two types of Google login exists
-        // TODO https://github.com/realm/realm-core/issues/5347
-        // assertFailsWith<IllegalArgumentException> { Credentials.google("", GoogleAuthType.AUTH_CODE) }
+        assertFailsWith<IllegalArgumentException> { Credentials.google("", GoogleAuthType.AUTH_CODE) }
         assertFailsWith<IllegalArgumentException> { Credentials.google("", GoogleAuthType.ID_TOKEN) }
     }
 
+    @Suppress("invisible_reference", "invisible_member")
```

Comment:
Not that you need to rework this, but I usually try to encapsulate usage of the internal API into a separate method, so that we don't accidentally unintentionally uses internal stuff elsewhere in the tests. Here you could do an `assertJson(....)` and do the cast and `asJson` in there.

Reply:
Good point 👍